nntrainer/optimizers/lr_scheduler.h

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2021 Parichay Kapoor <pk.kapoor@samsung.com>
+ *
+ * @file   lr_scheduler.h
+ * @date   09 December 2021
+ * @brief  This is Learning Rate Scheduler interface class
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Parichay Kapoor <pk.kapoor@samsung.com>
+ * @bug    No known bugs except for NYI items
+ *
+ */
+
+#ifndef __LEARNING_RATE_SCHEDULER__
+#define __LEARNING_RATE_SCHEDULER__
+#ifdef __cplusplus
+
+#include <string>
+
+#include <node_exporter.h>
+
+namespace nntrainer {
+
+/**
+ * @class   Learning Rate Schedulers Base class
+ * @brief   Base class for all Learning Rate Schedulers
+ */
+class LearningRateScheduler {
+
+public:
+  /**
+   * @brief     Destructor of learning rate scheduler Class
+   */
+  virtual ~LearningRateScheduler() = default;
+
+  /**
+   * @brief     get Learning Rate for the given iteration
+   * @param[in] iteration Iteration for the learning rate
+   * @retval    Learning rate in double
+   * @detail    the return value of this function and getInitialLearningRate()
+   * may not match for iteration == 0 (warmup can lead to different initial
+   * learning rates).
+   *
+   * @note this is non-const function intentionally.
+   */
+  virtual double getLearningRate(size_t iteration) = 0;
```

Comment:
the current interface does not have `finalize()` to check if the required properties are provided, etc and they throw in the first call to `getLearningRate()`. If you guys think `finalize()` would be better, I will add with interface `void finalize();` and it will throw if any error.

Reply:
+1 for this, this might cause problem some later :p

Reply:
added finalize with popular demand :)